test/test-files/step-function.ts

```diff
@@ -610,41 +610,36 @@ new StepFunction(stack, "sfn", async (input: { arr: string[] }) => {
 });
 
 /**
- * Unsupported - Filter
- * @see ErrorCodes.StepFunction_invalid_filter_syntax
+ * Unsupported
  */
 
 const someFunctionReference = () => {
   return true;
 };
 
+new StepFunction(stack, "fn", async () => {
+  [1, 2, 3].filter(someFunctionReference);
+  [1, 2, 3].map(someFunctionReference);
+  [1, 2, 3].forEach(someFunctionReference);
+});
```

Comment:
This can likely be supported now that we have enhanced parsing of all closures

Reply:
yeah, totally, needs to be tested